source/aerobulk/flux.py

```diff
@@ -9,29 +13,209 @@ def _check_algo(algo, valids):
+    evap : numpy.array
+        evaporation         [mm/s] aka [kg/m^2/s] (usually <0, as ocean loses water!)
+    """
+    (
+        ql,
+        qh,
+        taux,
+        tauy,
+        evap,
+    ) = aeronoskin.mod_aerobulk_wrapper_noskin.aerobulk_model_noskin(
+        algo, zt, zu, sst, t_zt, hum_zt, u_zu, v_zu, slp, niter
+    )
+    return ql, qh, taux, tauy, evap
+
+
+def skin_np(
+    sst,
+    t_zt,
+    hum_zt,
+    u_zu,
+    v_zu,
+    rad_sw,
+    rad_lw,
+    slp,
+    algo,
+    zt,
+    zu,
+    niter,
+):
+    """Python wrapper for aerobulk with skin correction.
+    !ATTENTION If input not provided in correct units, will crash.
+
+    Parameters
+    ----------
+    sst : numpy.array
+        Bulk sea surface temperature [K]
+    t_zt : numpy.array
+        Absolute air temperature at height zt [K]
+    hum_zt : numpy.array
+        air humidity at zt, can be given as:
+            - specific humidity                   [kg/kg]
+            - dew-point temperature                   [K]
+            - relative humidity                       [%]
+            => type should normally be recognized based on value range
+    u_zu : numpy.array
+        zonal wind speed at zu                    [m/s]
+    v_zu : numpy.array
+        meridional wind speed at zu               [m/s]
+    rad_sw : numpy.array
+        downwelling shortwave radiation at the surface (>0)   [W/m^2]
+    rad_lw : numpy.array
+        rad_lw : downwelling longwave radiation at the surface  (>0)   [W/m^2]
+    slp : numpy.array, optional
+        mean sea-level pressure                   [Pa] ~101000 Pa,
+        by default 101000.0
+    algo : str
+        Algorithm, can be one of: "coare3p0", "coare3p6", "ecmwf",
+    zt : int
+        height for temperature and spec. hum. of air           [m],
+    zu : int
+        height for wind (10m = traditional anemometric height  [m],
+    niter : int
+        Number of iteration steps used in the algorithm,
+
+    Returns
+    -------
+    ql : numpy.array
+        Latent heat flux                                     [W/m^2]
+    qh : numpy.array
+        Sensible heat flux                                   [W/m^2]
+    taux : numpy.array
+        zonal wind stress                                    [N/m^2]
+    tauy : numpy.array
+        meridional wind stress                                    [N/m^2]
+    t_s : numpy.array
+        skin temperature    [K]    (only when l_use_skin_schemes=TRUE)
+    evap : numpy.array
+        evaporation         [mm/s] aka [kg/m^2/s] (usually <0, as ocean loses water!)
+    """
+    (
+        ql,
+        qh,
+        taux,
+        tauy,
+        t_s,
+        evap,
+    ) = aeroskin.mod_aerobulk_wrapper_skin.aerobulk_model_skin(
+        algo, zt, zu, sst, t_zt, hum_zt, u_zu, v_zu, slp, rad_sw, rad_lw, niter
+    )
+    return ql, qh, taux, tauy, t_s, evap
+
+
+def input_check(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
```

Comment:
Doing this as a wrapper decorator is a nice idea, I like it.

Reply:
(Though technically the decorator checks both inputs and outputs, so it's an `input_and_output_check`)